tests/unit/test_http_client.py

```diff
@@ -223,3 +223,28 @@ def simple_json_callback(resp):
     assert exp_resp == resp
     assert 1 == mock_set_valid_token_called
     assert 2 == len(caplog.records)
+
+
+#####
+# Tests & fixtures the GPagintorMixin
+#####
+@pytest.mark.asyncio
+@pytest.mark.parametrize('max_pages', [1, 2])
+async def test_list_all(mocker, max_pages):
+    number_of_calls = 0
+
+    class TestClient(http_client.GPaginatorMixin):
+        async def get_json(self, url, params=None):
+            nonlocal number_of_calls
+            if number_of_calls < (max_pages - 1):
+                number_of_calls += 1
+                return {'data': 'data', 'nextPageToken': 'token'}
+            else:
```

Comment:
You actually don't need the `else` here since you have a `return` in the `if` clause. Here's [a good lightning talk](https://www.youtube.com/watch?v=JVVMMULwR4s) from Larry Hastings about it (starts around 4m50s).